pkg/api/handlers/compat/images_remove.go

```diff
@@ -30,28 +33,32 @@ func RemoveImage(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 	name := utils.GetName(r)
-	newImage, err := runtime.ImageRuntime().NewFromLocal(name)
-	if err != nil {
-		utils.ImageNotFound(w, name, errors.Wrapf(err, "failed to find image %s", name))
-		return
+	imageEngine := abi.ImageEngine{Libpod: runtime}
+
+	options := entities.ImageRemoveOptions{
+		Force: query.Force,
 	}
+	report, rmerrors := imageEngine.Remove(r.Context(), []string{name}, options)
+	if rmerrors != nil && rmerrors[0] != nil {
```

Comment:
This check can panic. `rmerrors` could be a empty slice (not nil) so access to the first element will panic. You would need to use `len(rmerrors) > 0`, this also covers the nil check.